### ISSUE TITLE
[CI] MixedClusterClientYamlTestSuiteIT tests failing against 8.0.0 after release

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#83904]
A whole slew of tests from `MixedClusterClientYamlTestSuiteIT` have started failing after the release of 8.0.0. You can see the [list of them here](https://gradle-enterprise.elastic.co/scans/tests?search.startTimeMax=1644860943864&search.startTimeMin=1644220800000&search.tasks=v8.0.0%23bwcTest&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT&tests.sortField=FAILED&tests.unstableOnly=true).

The errors differ but I think they all stem from some kind of transport layer incompatibility that breaks cluster formation.

```
» [2022-02-14T16:23:53,347][WARN ][o.e.t.InboundHandler     ] [v8.0.0-0] Failed to deserialize response from [127.0.0.1/127.0.0.1:39144]
»  org.elasticsearch.transport.TransportSerializationException: Failed to deserialize response from handler [org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler/org.elasticsearch.transport.TransportService$4/[indices:data/read/search[phase/query]]:org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler@5a467551/org.elasticsearch.action.search.SearchExecutionStatsCollector/org.elasticsearch.action.search.AbstractSearchAsyncAction$1@72f86089]
»  	at org.elasticsearch.transport.InboundHandler.handleResponse(InboundHandler.java:341) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:142) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:94) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:796) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:149) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:121) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:86) [elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:74) [transport-netty4-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280) [netty-handler-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [netty-transport-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [netty-common-4.1.73.Final.jar:4.1.73.Final]
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.73.Final.jar:4.1.73.Final]
»  	at java.lang.Thread.run(Thread.java:833) [?:?]
»  Caused by: java.lang.IllegalStateException: unexpected byte [0x06]
»  	at org.elasticsearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:563) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.StreamInput.readBoolean(StreamInput.java:553) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.StreamInput.readOptionalString(StreamInput.java:369) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.bucket.range.InternalRange.<init>(InternalRange.java:266) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:37) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:28) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.InternalAggregations.lambda$readFrom$1(InternalAggregations.java:59) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.StreamInput.readCollection(StreamInput.java:1155) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.StreamInput.readList(StreamInput.java:1106) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.aggregations.InternalAggregations.readFrom(InternalAggregations.java:59) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.DelayableWriteable.deserialize(DelayableWriteable.java:237) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.common.io.stream.DelayableWriteable.referencing(DelayableWriteable.java:57) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.query.QuerySearchResult.readFromWithId(QuerySearchResult.java:352) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.query.QuerySearchResult.<init>(QuerySearchResult.java:84) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.query.QuerySearchResult.<init>(QuerySearchResult.java:67) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.search.fetch.QueryFetchSearchResult.<init>(QueryFetchSearchResult.java:27) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:58) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:25) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TransportService$4.read(TransportService.java:734) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TransportService$4.read(TransportService.java:714) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1330) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1317) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	at org.elasticsearch.transport.InboundHandler.handleResponse(InboundHandler.java:338) ~[elasticsearch-8.2.0-SNAPSHOT.jar:8.2.0-SNAPSHOT]
»  	... 30 more
» [2022-02-14T16:23:53,355][ERROR][o.e.ExceptionsHelper     ] [v8.0.0-0] fatal error
»  	at org.elasticsearch.ExceptionsHelper.lambda$maybeDieOnAnotherThread$4(ExceptionsHelper.java:258)
»  	at java.base/java.util.Optional.ifPresent(Optional.java:178)
»  	at org.elasticsearch.ExceptionsHelper.maybeDieOnAnotherThread(ExceptionsHelper.java:248)
»  	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.exceptionCaught(Netty4MessageChannelHandler.java:81)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:302)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:381)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
»  	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:280)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
»  	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
»  	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
»  	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
»  	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
»  	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
»  	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623)
»  	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586)
»  	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
»  	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
»  	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
»  	at java.base/java.lang.Thread.run(Thread.java:833)
```

@salvatore-campagna mentions this being a [missing backport](https://github.com/elastic/elasticsearch/pull/83821) in 8.0. Since 8.0 released w/o that we'll need to fix things on the other end somehow.

**Comment by mark-vieira** [comment]
@salvatore-campagna can you confirm this should be addressed by https://github.com/elastic/elasticsearch/pull/83509? I just wanted to document this issue separately from https://github.com/elastic/elasticsearch/issues/83822 since they are unrelated.

**Comment by nik9000** [comment]
> @salvatore-campagna can you confirm this should be addressed by #83509? I just wanted to document this issue separately from #83822 since they are unrelated.

I can confirm that it *should* be addressed by that PR.

**Comment by nik9000** [comment]
And the time_series mode should be unrelated.

**Comment by salvatore-campagna** [comment]
Yes, I think that should be correct. I have just merged the PR to master. Backport PRs are being created. I will run them through a full bwc test before merging.

**Comment by salvatore-campagna** [comment]
For 8.1 there was a new build today at 14.55 after I merged the fix #83908 yesterday and it is green again (https://elasticsearch-ci.elastic.co/view/Elasticsearch%208.1/job/elastic+elasticsearch+8.1+periodic+bwc/BWC_VERSION=8.0.0,nodes=general-purpose/)

**Comment by mark-vieira** [comment]
This has been resolved by https://github.com/elastic/elasticsearch/pull/83909
